src/python_routing_v02/troute/routing/diffusive_utils.py

```diff
@@ -484,9 +494,47 @@ def diffusive_input_data_v02(
         if nnodes > mxncomp_g:
             mxncomp_g = nnodes
 
+
+    ds_seg = []
+    upstream_flow_array = np.zeros((len(ds_seg), np.shape(qlat_data)[1]))
+    if upstream_results:
+        # create a list of segments downstream of reservoirs
+        ds_seg = []
+        inv_map = nhd_network.reverse_network(rconn)
+        for wbody_id in upstream_results:
+            ds_seg.append(inv_map[wbody_id][0])
+        # build array of flow reservoir outflow
+        upstream_flow_array = np.zeros((len(ds_seg), np.shape(qlat_data)[1]))
+        for j, wbody_id in enumerate(upstream_results):
+            tmp = upstream_results[wbody_id]
+            for i, val in enumerate(tmp["results"][::3]):
+                if i%qts_subdivisions == 0:
+                    upstream_flow_array[j, int(i/qts_subdivisions)] = val
+                    
+        
+    
+    # edit rconn dict values to remove any values above reservoir tailwaters. 
+    # how to quickly find reservoir values in the rconn dictionary?
+        # search the connections dictionary for reservoir segments, that will give you the downstream segments
+        # for all connections in wbodies, give me the reverse connections
+        # will need to pass wbodies df (list)
+#     import pdb; pdb.set_trace()
+    
+    # attn - this is a hack for issues in dfs_decomposition_depth_tuple
+    # generalize these lines to replace rconn values for segs below reservoirs with [ ]
+    # need to pass waterbody list down to this level
+    
+    if 6227150 in rconn:
```

Comment:
I believe we are waiting on further confirmation of the network splitting before we get rid of this.